sql/task2/request-executor.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+include "./db-manipulation.php";
+
+define("MAX_SUBJECT_NAME_LENGTH", 128, true);
+define("MAX_LECTURE_NAME_LENGTH", 128, true);
+define("MAX_DESCRIPTION_LENGTH", 1024, true);
+define("MIN_DESCRIPTION_LENGTH", 10, true);
+
+$fileName = "request-executor.php";
+$dirname = basename($_SERVER['REQUEST_URI']);
+if ($_POST) {
+  if ($fileName == $dirname) {
+    executePostRequest(null);
+  }
+}
+
+function executePostRequest($electiveId)
+{
+  $subjectName = $_POST['subjectName'];
+  $lecturerName = $_POST['lecturerName'];
+  $description = $_POST['description'];
+  validateElectiveFields($subjectName, $lecturerName, $description);
+  echo("-----------");
+  if (is_null($electiveId)) {
+    createNewElective($subjectName, $lecturerName, $description);
+  } else {
+    echo("dasdasd2222");
+    updateExistingElective($subjectName, $lecturerName, $description, $electiveId);
+  }
+}
+
+function createNewElective($subjectName, $lecturerName, $description)
+{
+  addElactive($subjectName, $lecturerName, $description);
+  echo ("Elective is added sucesfully!");
+}
+
+function updateExistingElective($subjectName, $lecturerName, $description, $electiveId)
+{
+  checkIfElectiveExsists($electiveId);
+  updateElective($subjectName, $lecturerName, $description, $electiveId);
+  echo ("Elective is updated sucesfully!");
+}
+
+function getElective($electiveId)
+{
+  checkIfElectiveExsists($electiveId);
+
+  return getEletiveById($electiveId);
+}
+
+function checkIfElectiveExsists($electiveId)
+{
+  $elective = getEletiveById($electiveId);
+  $isElectiveExists = $elective["title"];
+  if (!$isElectiveExists) {
+    die("The elective with the given id: $electiveId is not existing.");
+  }
+}
+
+function validateElectiveFields($subjectName, $lecturerName, $description)
```

Comment:
You can combine all field validation functions in one function which can be reused.